src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs

```diff
@@ -150,12 +150,28 @@ public override void DeleteFile(string fullPath)
             if (Interop.Sys.Unlink(fullPath) < 0)
             {
                 Interop.ErrorInfo errorInfo = Interop.Sys.GetLastErrorInfo();
-                // ENOENT means it already doesn't exist; nop
-                if (errorInfo.Error != Interop.Error.ENOENT)
+                switch(errorInfo.Error)
                 {
-                    if (errorInfo.Error == Interop.Error.EISDIR)
+                    case Interop.Error.ENOENT:
+                        // ENOENT means it already doesn't exist; nop
+                        return;
+                    case Interop.Error.EROFS:
+                        // EROFS means the file system is read-only
+                        // Need to manually check file existence
+                        // github.com/dotnet/corefx/issues/21273
+                        Interop.ErrorInfo fileExistsError;
+
+                        // Windows doesn't care about the trailing separator
```

Comment:
We just kept the comment that was originally there in file.Exists, but the comment is explaining why the code trims the trailing separator (to match Windows behavior). 
If you have suggestions for a more clear way to say this we're open to suggestions.